src/seedu/addressbook/data/person/Person.java

```diff
@@ -134,8 +132,27 @@ public void removeAccount() {
         account = null;
     }
 
-    public List<Assessment> getAssessments() {
-        return new ArrayList<>(assessments);
+    public Set<Assessment> getAssessments() {
+        return new HashSet<>(assessments);
+    }
+
+    /**
+     * Checks if the assessment is already added
+     */
+    public boolean isAssessmentPresent(Assessment assessment) {
+        boolean present = false;
+        if (assessments.contains(assessment)) {
+            present = true;
+        }
+        return present;
+    }
```

Comment:
The copy pasta is real. I just realized the both of you have a mistakes
@micklwl 
The boolean variable should sound like a boolean.
Same as I told Mick, maybe it might be better just to return the value in the if statement to reduce LoC